Applications/Spire/Source/KeyBindings/OrderTasksPage.cpp

```diff
@@ -0,0 +1,805 @@
+      auto& side_rhs = any_cast<Side>(rhs);
+      if(side_lhs != Side::NONE && side_rhs != Side::NONE) {
+        return displayText(side_lhs) < displayText(side_rhs);
+      } else if(side_lhs != Side::NONE) {
+        return false;
+      } else if(side_rhs != Side::NONE) {
+        return true;
+      }
+    } else if(lhs.get_type() == typeid(TimeInForce)) {
+      return any_cast<TimeInForce>(lhs).GetType() <
+        any_cast<TimeInForce>(rhs).GetType();
+    } else if(lhs.get_type() == typeid(OrderType)) {
+      return any_cast<OrderType>(lhs) < any_cast<OrderType>(rhs);
+    } else if(lhs.get_type() == typeid(Destination)) {
+      return any_cast<Destination>(lhs) < any_cast<Destination>(rhs);
+    } else if(lhs.get_type() == typeid(Region)) {
+      return compare_text(display_region(any_cast<Region>(lhs)),
+        display_region(any_cast<Region>(rhs)));
+    } else if(lhs.get_type() == typeid(QKeySequence)) {
+      return compare_text(any_cast<QKeySequence>(lhs).toString(),
+        any_cast<QKeySequence>(rhs).toString());
+    }
+    return false;
+  }
+
+  auto is_in_layout(QLayout* layout, QWidget* widget) {
+    for(auto i = 0; i < layout->count(); ++i) {
+      if(layout->itemAt(i)->widget() == widget) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  QWidget& get_table_item_body(const TableItem& item) {
+    return static_cast<Button*>(item.layout()->itemAt(0)->widget())->get_body();
+  }
+
+  EditableBox* get_editable_box_from_item(const TableItem& item) {
+    return static_cast<EditableBox*>(&get_table_item_body(item));
+  }
+
+  EditableBox* get_editable_box_from_popup_item(const TableItem& item) {
+    return static_cast<EditableBox*>(&static_cast<PopupBox*>(
+      get_table_item_body(item).layout()->itemAt(0)->widget())->get_body());
+  }
+}
+
+std::size_t OrderTasksPage::RegionKeyHash::operator()(
+    const std::pair<Region, QKeySequence>& region_key) const {
+  auto seed = std::size_t(0);
+  hash_combine(seed, hash_range(region_key.first.GetCountries().begin(),
+    region_key.first.GetCountries().end()));
+  auto markets = region_key.first.GetMarkets();
+  hash_combine(seed, hash_range(markets.begin(), markets.end()));
+  hash_combine(seed, hash_range(region_key.first.GetSecurities().begin(),
+    region_key.first.GetSecurities().end()));
+  hash_combine(seed, qHash(region_key.second));
+  return seed;
+}
+
+OrderTasksPage::AddedRow::AddedRow()
+  : m_source_index(-1),
+    m_is_filtered(false) {}
+
+OrderTasksPage::OrderTasksPage(
+    std::shared_ptr<ComboBox::QueryModel> region_query_model,
+    std::shared_ptr<ListModel<OrderTask>> model,
+    const DestinationDatabase& destination_database,
+    const MarketDatabase& market_database, QWidget* parent)
+    : QWidget(parent),
+      m_region_query_model(std::move(region_query_model)),
+      m_destination_database(destination_database),
+      m_market_database(market_database),
+      m_model(std::move(model)) {
```

Comment:
The order of initialization doesn't match the order of the parameters. Look into this further.